src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -231,13 +240,52 @@ private void RunChildren()
 
         private void CreateChildWorkItems()
         {
-            _children = new List<WorkItem>();
+            Children = new List<WorkItem>();
 
             foreach (ITest test in _suite.Tests)
                 if (_childFilter.Pass(test))
-                    _children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                {
+                    if (test.Properties.ContainsKey(PropertyNames.Order))
+                    {
+                        Children.Insert(0, WorkItem.CreateWorkItem(test, _childFilter));
+                        _countOrder++;
+                    }
+                    else
+                    {
+                        Children.Insert(Children.Count, WorkItem.CreateWorkItem(test, _childFilter));
```

Comment:
I pushed for this, not because of optimization, but so we would not change the existing order of non-ordered items.

An alternative would be the two-loop approach that @dicko2 originally used: first add the ordered items, then add the unordered items. Then sort the ordered items only. Alternatively, do the sort before adding the unordered items! If List.Sort were stable (retained pre-existing order for equal items) then we wouldn't have to do this.